### ISSUE TITLE
0.11.5 breaks repeat using `$data`

### ISSUE THREAD

**Issue by benceg** [yyx990803/vue#707]
I'm writing an application that needs to list arbitrary values in a JSON object two levels deep. Some of these values are arrays whose constituent objects contain `id` and `value` properties.

Consider the following code:

```js
...
data : function() {
  return {
    model : {
      "arbitrary_key" : [
        { "id" : "1", "value" : "arbitrary value 1" },
        { "id" : "2", "value" : "arbitrary value 2" },
        { "id" : "3", "value" : "arbitrary value 3" },
        { "id" : "4", "value" : "arbitrary value 4" }
      ]
    }
  }
}
...
```

```html
<li v-repeat="model">
  
  {{ $key }}
  
  <ul>
    <li v-repeat="$data">
      <label>{{ id }}</label>
      <var>{{ value }}</var>
    </li>
  </ul>
  
</li>
```

The code above outputs the list of four items as expected using Vue 0.11.4, but outputs nothing when using Vue 0.11.5. I don't believe there is a test case to cover this particular use case.

**Comment by TMiguelT** [comment]
Can you show me a JSFiddle that shows this behaviour? I copied your code and changed it from a component to a base Vue instance and it's completely skipping the `v-repeat=$data`

http://jsfiddle.net/o9j3yjcq/

**Comment by calebboyd** [comment]
I think thats the point. You can change to 0.11.4 and it renders as expected. Something to do with:
https://github.com/yyx990803/vue/commit/ab254f7e8faeeef697445e6bdd622d7bcaa316a1

**Comment by benceg** [comment]
Apologies, @TMiguelT, my post was unnecessarily ambiguous! I skipped the bit where my template spews out `$value` if `$data | empty` (a custom filter) returns `true`.

So expected behaviour was actually to see nothing; but in my case it was displaying `$value` - a stringified array of objects. I've updated the OP to avert any possible confusion for anyone else who might have caught the same issue.

Thank you so much for the fix @calebboyd.
